Tests/FluentAssertions.Specs/Events/EventMonitorSpecs.cs

```diff
@@ -0,0 +1,143 @@
+﻿#if NETFRAMEWORK
+using System.Reflection.Emit;
+#endif
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+#pragma warning disable CS0067
+
+namespace FluentAssertions.Specs.Events;
+
+[Collection("EventMonitoring")]
+public class EventMonitorSpecs
+{
+    public class MonitorDefaultBehavior
+    {
+        [Fact]
+        public void When_default_configuration_is_active_broken_event_add_accessors_should_fail_the_test()
+        {
+            // Arrange
+            var cut = new TestEventBrokenEventHandlerRaising();
+
+            // Act
+            // Assert
+            cut.Invoking(c =>
+            {
+                using var monitor = c.Monitor<IAddFailingEvent>();
+            }).Should().Throw<TargetInvocationException>();
+        }
+
+        [Fact]
+        public void When_default_configuration_is_active_broken_event_remove_accessors_should_fail_the_test()
+        {
+            // Arrange
+            var cut = new TestEventBrokenEventHandlerRaising();
+
+            // Act
+            // Assert
+            cut.Invoking(c =>
+            {
+                using var monitor = c.Monitor<IRemoveFailingEvent>();
+            }).Should().Throw<TargetInvocationException>();
+        }
+    }
+
+    public class IgnoreMisbehavingEventAccessors
+    {
+        [Fact]
+        public void
+            When_IgnoreEventAccessorExceptions_is_set_monitoring_class_with_broken_event_add_accessor_should_not_fail_test()
+        {
+            // Arrange
+            var classToMonitor = new TestEventBrokenEventHandlerRaising();
+
+            //Act
+            //Assert
+            classToMonitor.Invoking(c =>
+            {
+                using var monitor = c.Monitor<IAddFailingEvent>(opt => opt.IgnoreEventAccessorExceptions());
+            }).Should().NotThrow();
+        }
+
+        [Fact]
+        public void
+            When_IgnoreEventAccessorExceptions_is_set_monitoring_class_with_broken_event_remove_accessor_should_not_fail_test()
+        {
+            // Arrange
+            var classToMonitor = new TestEventBrokenEventHandlerRaising();
+
+            //Act
+
+            //Assert
+            classToMonitor.Invoking(c =>
+            {
+                using var monitor = c.Monitor<IRemoveFailingEvent>(opt => opt.IgnoreEventAccessorExceptions());
+            }).Should().NotThrow();
+        }
+    }
+
+    public class SettingTimestampProvider
```

Comment:
🔧 These tests are not needed. The only reason why we allow passing a custom date/time provider is so that we can decouple the tests from date/time issues.

Reply:
done